extractors/rumble/rumble.go

```diff
@@ -0,0 +1,267 @@
+package rumble
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/pkg/errors"
+
+	"github.com/iawia002/lux/extractors"
+	"github.com/iawia002/lux/request"
+	"github.com/iawia002/lux/utils"
+)
+
+func init() {
+	extractors.Register("rumble", New())
+}
+
+type extractor struct{}
+
+// New returns a rumble extractor.
+func New() extractors.Extractor {
+	return &extractor{}
+}
+
+type rumbleData struct {
+	Format       string `json:"format"`
+	Name         string `json:"name"`
+	EmbedURL     string `json:"embedUrl"`
+	ThumbnailURL string `json:"thumbnailUrl"`
+	Type         string `json:"@type"`
+	VideoURL     string `json:"videoUrl"`
+	Quality      string `json:"quality"`
+}
+
+// Extract is the main function to extract the data.
+func (e *extractor) Extract(url string, option extractors.Options) ([]*extractors.Data, error) {
+	res, err := request.Request(http.MethodGet, url, nil, nil)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	defer res.Body.Close() // nolint
+
+	var reader io.ReadCloser
+	switch res.Header.Get("Content-Encoding") {
+	case "gzip":
+		reader, _ = gzip.NewReader(res.Body)
+	case "deflate":
+		reader = flate.NewReader(res.Body)
+	default:
+		reader = res.Body
+	}
+	defer reader.Close() // nolint
+
+	b, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	html := string(b)
+
+	cookiesArr := make([]string, 0)
+	cookies := res.Cookies()
+
+	for _, c := range cookies {
+		cookiesArr = append(cookiesArr, c.Name+"="+c.Value)
```

Comment:
`cookiesArr` is never used, do we need this?

Reply:
Right, I'll make a couple more changes.